libnmstate/nm/bridge_port_vlan.py

```diff
@@ -0,0 +1,252 @@
+            return False
+
+    @staticmethod
+    def _get_vlan_info(port_vlan_info):
+        vlan_min, vlan_max = PortVlanFilter.get_vlan_tag_range(port_vlan_info)
+        pvid = port_vlan_info.is_pvid()
+        untagged = port_vlan_info.is_untagged()
+        if vlan_max != vlan_min:
+            port_data = {
+                LB.Port.Vlan.TrunkTags.ID_RANGE: {
+                    LB.Port.Vlan.TrunkTags.MIN_RANGE: vlan_min,
+                    LB.Port.Vlan.TrunkTags.MAX_RANGE: vlan_max,
+                }
+            }
+        else:
+            port_data = {LB.Port.Vlan.TrunkTags.ID: vlan_min}
+        port_data.update(pvid=pvid, untagged=untagged)
+        return port_data
+
+    @staticmethod
+    def _generate_vlan_trunk_port_config(trunk_port):
+        single_vlan_tag = trunk_port.get(LB.Port.Vlan.TrunkTags.ID)
+        ranged_vlan_tags = trunk_port.get(LB.Port.Vlan.TrunkTags.ID_RANGE)
+        port_vlan = None
+
+        if single_vlan_tag:
+            port_vlan = nmclient.NM.BridgeVlan.new(
+                single_vlan_tag, single_vlan_tag
+            )
+        elif ranged_vlan_tags:
+            min_range, max_range = (
+                ranged_vlan_tags[range_type]
+                for range_type in (
+                    LB.Port.Vlan.TrunkTags.MIN_RANGE,
+                    LB.Port.Vlan.TrunkTags.MAX_RANGE,
+                )
+            )
+            port_vlan = nmclient.NM.BridgeVlan.new(min_range, max_range)
+        port_vlan.set_untagged(False)
+        port_vlan.set_pvid(False)
+        return port_vlan
+
+    @staticmethod
+    def _generate_vlan_access_port_config(vlan_tag):
+        port_vlan = nmclient.NM.BridgeVlan.new(vlan_tag, vlan_tag)
+        port_vlan.set_untagged(True)
+        port_vlan.set_pvid(True)
+        return port_vlan
+
+    @staticmethod
+    def _get_vlan_config(bridge_vlan_config):
+        enable_native = False
+        trunk_tags = []
+        tag = None
+
+        for port_vlan_filter_entry in bridge_vlan_config:
+            single_vlan_tag = port_vlan_filter_entry.get(
+                LB.Port.Vlan.TrunkTags.ID
+            )
+            ranged_vlan_tags = port_vlan_filter_entry.get(
+                LB.Port.Vlan.TrunkTags.ID_RANGE
+            )
+            if PortVlanFilter._is_access_port(bridge_vlan_config):
+                tag = single_vlan_tag
+            else:
+                if port_vlan_filter_entry[PortVlanFilter.Pvid]:
+                    tag = (
+                        single_vlan_tag
+                        or ranged_vlan_tags[LB.Port.Vlan.TrunkTags.MAX_RANGE]
+                    )
+                    enable_native = True
+                else:
+                    PortVlanFilter._sanitize_trunk_port(port_vlan_filter_entry)
+                    trunk_tags.append(port_vlan_filter_entry)
+
+        return trunk_tags, tag, enable_native
+
+    @staticmethod
+    def _sanitize_trunk_port(port):
+        if PortVlanFilter.Pvid and PortVlanFilter.Untagged in port:
```

Comment:
The `_sanitize_trunk_port()` is only used in `_get_vlan_config()`, at its invoke time, it already checked the ` if port_vlan_filter_entry[PortVlanFilter.Pvid]`.

Since `_sanitize_trunk_port()` is small, please just merge it into `_get_vlan_config()`.

Reply:
Aye aye captain.

Done.